src/lib/merkle_ledger/converting_merkle_tree.ml

```diff
@@ -0,0 +1,211 @@
+open Core_kernel
```

Comment:
This is all Matthew's original work. The only changes I made were to use `Option` combinators rather than explicitly matching on `Some/None`